tests/init/neg/local-warm.scala

```diff
@@ -0,0 +1,16 @@
+abstract class A {
+    def m() = 10
+    def m1(): B = new B
+    def m2(): Int = m1().m()
+    class B extends A {
+        def x = 10
+    }
+}
+
+class C extends A {
+    def g() = {
+        val t = m1() // error
```

Comment:
Opportunity for improvement: for local definitions (not params), we can re-compute its value from its `rhs`. There is no need to require it be `hot`.

Reply:
i've fixed that